src/lxc/criu.c

```diff
@@ -439,27 +439,53 @@ static void exec_criu(struct criu_opts *opts)
 			char eth[128], *veth;
 			struct lxc_netdev *n = it->elem;
 
-			if (n->type != LXC_NET_VETH)
-				continue;
-
 			if (n->name) {
 				if (strlen(n->name) >= sizeof(eth))
 					goto err;
 				strncpy(eth, n->name, sizeof(eth));
 			} else
 				sprintf(eth, "eth%d", netnr);
 
-			veth = n->priv.veth_attr.pair;
+			switch (n->type) {
+			case LXC_NET_VETH:
+				veth = n->priv.veth_attr.pair;
 
-			if (n->link)
-				ret = snprintf(buf, sizeof(buf), "%s=%s@%s", eth, veth, n->link);
-			else
-				ret = snprintf(buf, sizeof(buf), "%s=%s", eth, veth);
-			if (ret < 0 || ret >= sizeof(buf))
+				if (n->link)
+					ret = snprintf(buf, sizeof(buf), "veth[%s]:%s@%s", eth, veth, n->link);
+				else
+					ret = snprintf(buf, sizeof(buf), "veth[%s]:%s", eth, veth);
+				if (ret < 0 || ret >= sizeof(buf))
+					goto err;
+				break;
+			case LXC_NET_MACVLAN:
+				if (n->name) {
+					if (strlen(n->name) >= sizeof(eth))
+						goto err;
+					strncpy(eth, n->name, sizeof(eth));
+				} else
+					sprintf(eth, "eth%d", netnr);
```

Comment:
Actually we can just drop this hunk entirely, as it's done above (also without a snprintf... I'll push patches for both).